File: qualification-round/q2/q2-solution.go
package main

import (
	"fmt"
)

func main() {

	// Scan first line
	var input int
	var gridTemp int
	var temp string
	fmt.Scanln(&input)

	grid := []int{}
	array := []string{}
	for i := 0; i < input; i++ {
		fmt.Scanln(&temp)
		grid = append(grid, gridTemp)
		fmt.Scanln(&temp)
		array = append(array, temp)
	}

	// Check for input
	for i := range array {
		answer := mirrorLydiaSolution(array[i])
		fmt.Printf("Case #%d: %s\n", i+1, answer)
	}
}

func mirrorLydiaSolution(lydia string) string {
	solution := ""
	buffer := ""
	moveToMake := ""
	for i, c := range lydia {
		if moveToMake != mirrorMove(c) {
			solution += buffer
			buffer = ""
		}
		moveToMake = mirrorMove(c)
		buffer += moveToMake
		if i == len(lydia)-1 {
			solution += buffer
		}
	}
	return solution
}

func mirrorMove(move rune) string {
	if move == 'S' {
		return "E"
	} else if move == 'E' {
		return "S"
	} else {
		return " "
	}
}

File: qualification-round/q2/q2-solution.py
#!/bin/env/python3

def mirror(x):
    if x == "E":
        return "S"
    else:
        return "E"

# input() reads a string with a line of input, stripping the ' ' (newline) at the end.
# This is all you need for most Code Jam problems.
t = int(input()) # read a line with a single integer
for i in range(1, t + 1):

    n = input() # read a list of integers, 2 in this case
    path = input() # read the path

    answer = ""
    buffer = 0
    firstchar = ""

    #
    for j in range(0, len(path)):
        answer += mirror(path[j])

    # check out .format's specification for more formatting options
    print("Case #{}: {}".format(i, answer))